plugins/post-transaction-actions.py

```diff
@@ -0,0 +1,158 @@
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+"""
+Plugin runs actions (shell commands) after transaction is completed.
+Actions are defined in "*.action" files.
+The same shell command is executed only once.
+"""
+
+from dnfpluginscore import _, logger
+
+import libdnf.conf
+import dnf
+import dnf.transaction
+import glob
+import os
+import subprocess
+
+
+class PostTransactionActions(dnf.Plugin):
+
+    name = "post-transaction-actions"
+
+    def __init__(self, base, cli):
+        super(PostTransactionActions, self).__init__(base, cli)
+        self.actiondir = "/etc/dnf/plugins/post-transaction-actions.d/"
+        self.base = base
+        self.logger = logger
+
+    def config(self):
+        conf = self.read_config(self.base.conf)
+        if conf.has_section("main"):
+            if conf.has_option("main", "actiondir"):
+                self.actiondir = conf.get("main", "actiondir")
+
+    def _parse_actions(self):
+        """Parses *.action files from self.actiondir path.
+        Parsed actions are stored in a list of tuples."""
+
+        action_file_list = []
+        if os.access(self.actiondir, os.R_OK):
+            action_file_list.extend(glob.glob(self.actiondir + "*.action"))
+
+        action_tuples = []  # (action key, action_state, shell command)
+        for file_name in action_file_list:
+            for line in open(file_name).readlines():
+                line = line.strip()
+                if line and line[0] != "#":
+                    try:
+                        (action_key, action_state, action_command) = line.split(':', 2)
+                    except ValueError as e:
+                        self.logger.error(_('Bad Action Line "%s": %s') % (line, e))
+                        continue
+                    else:
+                        action_tuples.append((action_key, action_state, action_command))
+
+        return action_tuples
+
+    def _replace_vars(self, ts_item, command):
+        """Replaces variables in the command.
+        Variables: ${name}, ${arch}, ${ver}, ${rel}, ${epoch}, ${repoid}, ${state}
+        or $name, $arch, $ver, $rel, $epoch, $repoid, $state"""
+
+        action_dict = {dnf.transaction.PKG_DOWNGRADE: "downgrade",
+                       dnf.transaction.PKG_DOWNGRADED: "downgraded",
+                       dnf.transaction.PKG_INSTALL: "install",
+                       dnf.transaction.PKG_OBSOLETE: "obsolete",
+                       dnf.transaction.PKG_OBSOLETED: "obsoleted",
+                       dnf.transaction.PKG_REINSTALL: "reinstall",
+                       dnf.transaction.PKG_REINSTALLED: "reinstalled",
+                       dnf.transaction.PKG_REMOVE: "remove",
+                       dnf.transaction.PKG_UPGRADE: "upgrade",
+                       dnf.transaction.PKG_UPGRADED: "upgraded"}
+
+        try:
+            action = action_dict[ts_item.action]
+        except KeyError:
+            action = "unknown - %s" % ts_item.action
```

Comment:
This could be shortened using .get() method and default:
```
action = action_dict.get(ts_item.action, "unknown - %s" % ts_item.action)
```

Reply:
Yes. It looks better. I will change it.